code/game/objects/structures/industrial_lift.dm

```diff
@@ -292,6 +292,9 @@ GLOBAL_LIST_EMPTY(lifts)
 			var/datum/callback/land_slam = new(collided, /mob/living/.proc/tram_slam_land)
 			collided.throw_at(throw_target, 200, 4, callback = land_slam)
 
+		for(var/atom/movable/specific_items as anything in destination.contents)
+			SEND_SIGNAL(specific_items, COMSIG_ATOM_INDUSTRIAL_LIFT_SPECIAL)
```

Comment:
`travel()` already has quite a few for loops. I believe this could be improved by making it a turf signal, which can then be connected to movables through the connect_loc element and its component counterpart.

Also why is the signal called `COMSIG_ATOM_INDUSTRIAL_LIFT_SPECIAL` when the proc is named `travel`?

Reply:
you vastly over-estimate how much i understand the sentence you just typed

the lift itself already uses connect_loc to handle adding and removing items from the list of things to move and because of that i honestly have no idea how to do it the way you're asking

also the travel proc handles also hitting things in the way of traveling so i figured it made sense

Reply:
> you vastly over-estimate how much i understand the sentence you just typed

Fuck, I guess i'll have to make myself clear then.

The signal should be send to the turf. Any movable atom that has some special interaction for when the tram bumps into or runs over it can have a connect_loc element (or a connect_loc_behalf component if it's a datum and/or is registering signals on behalf of another movable) attached to it (You can take a look at the documentation on the connect_loc element if you have little idea of what it's and how it works). Basically, you'd be sending the signal no more than one time per `/obj/structure/industrial_lift/travel()` call.

Reply:
i read this and had a brain blast and for whatever reason loc_connections made sense thanks